motionplan/motionPlanner.go

```diff
@@ -54,10 +54,49 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// PlanRequestValidation ensures PlanRequests are not malformed.
+func PlanRequestValidation(req PlanRequest) (*PlanRequest, error) {
```

Comment:
Making this public implies the user will be responsible for calling that. I think this probably should be private and just taken care of by the motionplan endpoints that take plan requests. 

Also would prefer it as a method on the struct